pyaerocom/colocation/colocated_data.py

```diff
@@ -856,38 +856,31 @@ def check_set_countries(self, inplace=True, assign_to_dim=None):
             name of dimension to which the country coordinate is assigned.
             Default is None, in which case station_name is used.
 
-        Raises
-        ------
-        DataDimensionError
-            If data is 4D (i.e. if latitude and longitude are othorgonal
-            dimensions)
-
         Returns
         -------
         ColocatedData
             data object with countries assigned
 
         """
-        if self.has_latlon_dims:
-            raise DataDimensionError(
-                "Countries cannot be assigned to 4D"
-                "ColocatedData with othorgonal lat / lon "
-                "dimensions. Please consider stacking "
-                "the latitude and longitude dimensions-"
-            )
         if assign_to_dim is None:
             assign_to_dim = "station_name"
 
-        if assign_to_dim not in self.dims:
-            raise DataDimensionError("No such dimension", assign_to_dim)
+        # stack lat and lon dimensions into station_name for 4D data
+        if all(_ in self.dims for _ in ("latitude", "longitude")):
```

Comment:
I don't like a underscore as meaningful loop variable. Underscore should only be used if it isn't used at another place. It is to easy to overlook an undescore. I would use `dim` or `d` here as understandable variable name.